### PR TITLE
2069 geocoder results

### PR DESCRIPTION
**Pull request by chiatt**
### Description of Change
Replaced geocoder proxy classes with knockout components.
Added a django model for geocoders and made it available in the admin interface.
Switched the default geocoder to mapbox.

### CODE REVIEW

File: arches/app/media/js/views/components/geocoders/mapzen.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+define(['knockout', 'mapbox-gl', 'arches', 'views/components/geocoders/base-geocoder', 'geocoder-templates'],
+function (ko, mapboxgl, arches) {
+    return ko.components.register('views/components/geocoders/mapzen', {
+        viewModel: function(params) {
+            BaseGeocoderViewModel.apply(this, [params]);
+            var self = this;
+            this.placeholder = params.placeholder || ko.observable('Locate a Place or Address');
+            this.anchorLayerId = params.anchorLayerId;
+            this.apiKey = params.api_key || arches.mapboxApiKey
```
Comment (robgaston) [inline arches/app/media/js/views/components/geocoders/mapzen.js]: @chiatt Shouldn't this reference `arches.mapzenApiKey`?

File: arches/app/templates/views/components/geocoders/mapbox.htm [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{% load i18n %}
```
Comment (robgaston) [inline arches/app/templates/views/components/geocoders/mapbox.htm]: I think that these two templates (`mapbox.htm` and `mapzen.htm`) are identical, no? I think we could just create a single `geocoder.htm` template that both of these components could use.

File: arches/app/media/js/views/components/geocoders/mapzen.js (hunk 1 above)
Comment (chiatt) [inline arches/app/media/js/views/components/geocoders/mapzen.js]: Ah - good catch

File: arches/app/templates/views/components/geocoders/mapbox.htm (hunk 2 above)
Comment (chiatt) [inline arches/app/templates/views/components/geocoders/mapbox.htm]: Yeah - I think they are. I'll go ahead and combine them.

**Comment by chiatt** [comment]
@robgaston Ok - I've consolidated the geocoder templates into `geocoder.htm`.

**Pull request merged**